### PR TITLE
add admitPod

### PR DESCRIPTION
**Pull request by wangyuqing4**
Fixes: #135 

Add admission controller(admitPod)

add ValidatingWebhookConfiguration volcano-validate-pod, only limit CREATE pods, allow pods to create when:
- pod.spec.schedulerName is not volcano
- podgroup phase isn't Pending
- normal job, no podgroup

### CODE REVIEW

**Comment by TommyLike** [comment]
@k82cn  @hzxuzhonghu  is this ready to go?

**Comment by hzxuzhonghu** [comment]
We need description about what it does.  So it is easy for us to trace new features on release note.

**Comment by k82cn** [comment]
@hzxuzhonghu @TommyLike , please help to review :)

File: cmd/admission/main.go [hunk 1]
```diff
@@ -63,7 +64,9 @@ func main() {
 		glog.Fatalf("Unable to build k8s config: %v\n", err)
 	}
 
-	admissioncontroller.KubeBatchClientSet = app.GetKubeBatchClient(restConfig)
+	admissioncontroller.VolcanoClientSet = app.GetVolcanoClient(restConfig)
+
+	servePods(config)
```
Comment (hzxuzhonghu) [inline cmd/admission/main.go]: Why donot you keep consistent with L54-55?

File: pkg/admission/admit_pod.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+		return pod, err
+	}
+	glog.V(3).Infof("the pod struct is %+v", pod)
+
+	return pod, nil
+}
+
+// allow pods to create when
+// 1. pod.spec.schedulerName != volcano
+// 2. Podgroup phase isn't Pending
```
Comment (hzxuzhonghu) [inline pkg/admission/admit_pod.go]: add more comments why do this?

**Comment by hzxuzhonghu** [comment]
Actually, i am not sure why need to delay pod creation? Can you elaborate a little bit more?

File: pkg/admission/admit_pod.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+		if err := c.checkPGPhase(pod, pgName, true); err != nil {
+			msg = err.Error()
+			reviewResponse.Allowed = false
+		}
+		return msg
+	}
+
+	// normal pod, SN == volcano
+	pgName = helpers.GeneratePodgroupName(&pod)
+	if pgName != "" {
```
Comment (TommyLike) [inline pkg/admission/admit_pod.go]: when this will happen?
```
if pgName != "" {
}
```

File: pkg/admission/admit_pod.go [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+	if err != nil {
+		if isVCJob || (!isVCJob && !apierrors.IsNotFound(err)) {
+			return fmt.Errorf("Failed to get PodGroup for pod <%s/%s>: %v", pod.Namespace, pod.Name, err)
+		}
+		return nil
+	}
+	if pg.Status.Phase != v1alpha2.PodGroupPending {
+		return nil
+	}
+	return fmt.Errorf("Failed to create pod for pod <%s/%s>, because the podgroup phase is Pending",
```
Comment (TommyLike) [inline pkg/admission/admit_pod.go]:
```
Failed to create pod <%s/%s>
```

File: cmd/admission/main.go (hunk 1 above)
Comment (wangyuqing4) [reply cmd/admission/main.go]: because pods need client and parameter。

File: pkg/admission/admit_pod.go (hunk 3 above)
Comment (wangyuqing4) [reply pkg/admission/admit_pod.go]: em，i will delete this if

**Pull request merged**